packages/components/src/internal/components/entities/actions.ts

```diff
@@ -1269,3 +1269,44 @@ export function getOrderedSelectedMappedKeysFromQueryModel(
         viewName
     );
 }
+
+function getDataTypeDataExistSql(dataType: ProjectConfigurableDataType, lsid?: string, rowId?: number): string {
+    if (!dataType) return null;
+
+    let from = 'exp.materials ';
+    // samples and assay runs reference their data type by lsid, but dataclass data reference by rowid
+    let where = "WHERE SampleSet = '" + lsid + "'";
+
+    if (dataType === 'DataClass') {
+        from = 'exp.data ';
+        where = 'WHERE DataClass = ' + rowId;
+    } else if (dataType === 'AssayDesign') {
+        from = 'exp.AssayRuns ';
+        where = "WHERE protocol = '" + lsid + "'";
+    }
+
+    return `SELECT count(*) as HasData FROM (SELECT 1 FROM ${from} ${where} limit 1)`;
+}
+
+export function isDataTypeEmpty(
+    dataType: ProjectConfigurableDataType,
+    lsid?: string,
+    rowId?: number,
+    containerPath?: string
+): Promise<boolean> {
+    return new Promise((resolve, reject) => {
+        Query.executeSql({
+            containerPath,
+            containerFilter: Query.ContainerFilter.allInProject,
+            schemaName: SCHEMAS.EXP_TABLES.SCHEMA,
+            sql: getDataTypeDataExistSql(dataType, lsid, rowId),
```

Comment:
interesting approach with the executeSQL call. I was expecting to see just a standard selectRows call against the given table (i.e. sample type, data class, etc) here with a maxRows=1. But I suppose that technically the default view for that table could have a filter on it that makes it look empty. Was that part of the thought here with using executeSQL and filtering by lsid/rowId?

Reply:
For large tables, selectRows with maxRows=1 would be slow. This check should be much faster.